### DISCUSSION TITLE
Which AI model does GitButler use?

### DISCUSSION THREAD

**Discussion by lkaino** [gitbutlerapp/gitbutler#2706] [Q&A] [answered]
I couldn't find this information in the documentation. Which LLM model is the app using? Is it local or cloud based? Is my data secure?

**Comment by krlvi** [comment, marked as answer]
Hi @lkaino! Got this question earlier today in a gh issue here: https://github.com/gitbutlerapp/gitbutler/issues/2701
I will copy paste my answer from there:

>The app currently uses OpenAI, but we would love to make it pluggable so that people can choose their own endpoint, configure the prompt etc. According to @pngwn it should be even possible to do this with the newest local models that have recently come out.

Just merged this pr which clarifies it in the readme and the app copy as well:
https://github.com/gitbutlerapp/gitbutler/pull/2707

Separately but on a related note there is a ticket tracking the support for customising the LLM endpoint https://github.com/gitbutlerapp/gitbutler/issues/2660 as well as customising the prompt https://github.com/gitbutlerapp/gitbutler/issues/2624

According to @pngwn it should be even possible to do this with the newest local models that have recently come out. I would love to try setting this up!

**Reply by lkaino** [reply]
Thank you for the good and clear answer! A local LLM support would be awesome from data security point of view.
